### PR TITLE
ProposerVM Extend windows 1 - UTs Cleanup

### PR DESCRIPTION
**Pull request by abi87**
## Why this should be merged
A few proposerVM unit tests are brittle.

- They may assert the wrong error type
- They make reference to core blocks Timestamps, while proposerVM production code never reference inner blocks timestamps post activation.
- They may not properly set core blocks Height and Parent attributes, while proposerVM production code rely on them

This PR fixes these issues.

## How this works
Just fixed the UTs

## How this was tested
Manual inspection of each UT

### CODE REVIEW

File: vms/proposervm/post_fork_block_test.go [hunk 1]
```diff
@@ -131,62 +131,64 @@ func TestBlockVerify_PostForkBlock_ParentChecks(t *testing.T) {
...
-	require.NoError(prntProBlk.Verify(context.Background()))
-	require.NoError(proVM.SetPreference(context.Background(), prntProBlk.ID()))
+	require.NoError(parentBlk.Verify(context.Background()))
+	require.NoError(proVM.SetPreference(context.Background(), parentBlk.ID()))
 
 	// .. create child block ...
 	childCoreBlk := &snowman.TestBlock{
-		ParentV:    prntCoreBlk.ID(),
-		BytesV:     []byte{2},
-		TimestampV: prntCoreBlk.Timestamp(),
```
Comment (abi87) [inline vms/proposervm/post_fork_block_test.go:152]: this the core block of a would be post fork block, its timestamp is not relevant to proposerVM. It's height and parent may be. So we better set up the right attributes

File: vms/proposervm/post_fork_block_test.go [hunk 2]
```diff
@@ -131,62 +131,64 @@ func TestBlockVerify_PostForkBlock_ParentChecks(t *testing.T) {
...
+		ParentV: parentCoreBlk.ID(),
+		BytesV:  []byte{2},
+		HeightV: parentCoreBlk.Height() + 1,
 	}
-	childSlb, err := block.Build(
```
[the remark this answers is not in the document]
Comment (abi87) [reply vms/proposervm/post_fork_block_test.go:154]: Ideally we'd like the same check carried out for signed blocks. We may add these UTs in a subsequent PR

File: vms/proposervm/post_fork_block_test.go [hunk 3]
```diff
@@ -131,62 +131,64 @@ func TestBlockVerify_PostForkBlock_ParentChecks(t *testing.T) {
...
-	childSlb, err = block.BuildUnsigned(
-		prntProBlk.ID(), // refer known parent
-		prntProBlk.Timestamp().Add(proposer.MaxVerifyDelay),
-		pChainHeight,
-		childCoreBlk.Bytes(),
-	)
-	require.NoError(err)
-	childProBlk.SignedBlock = childSlb
-	require.NoError(err)
+	{
```
Comment (abi87) [inline vms/proposervm/post_fork_block_test.go:164]: I scoped different tests cases this way, as we do in other parts of the codebase. Makes this diff more annoying but I think it improves overall readability. Also the more tightly scoped variables, the easier the maintanability I think

File: vms/proposervm/post_fork_block_test.go [hunk 4]
```diff
@@ -401,102 +428,114 @@ func TestBlockVerify_PostForkBlock_PChainHeightChecks(t *testing.T) {
...
 		},
-		ParentV:    prntCoreBlk.ID(),
-		BytesV:     []byte{2},
-		TimestampV: prntProBlk.Timestamp().Add(proposer.MaxVerifyDelay),
+		ParentV: parentCoreBlk.ID(),
+		BytesV:  []byte{2},
 	}
-
-	// child P-Chain height must not precede parent P-Chain height
-	childSlb, err := block.Build(
```
Comment (abi87) [inline vms/proposervm/post_fork_block_test.go:430]: once again a signed block and some unsigned blocks after. Not sure why. Moved to all unsigned blocks. To be revisited to extend UTs coverage over signed block.

File: vms/proposervm/post_fork_block_test.go [hunk 5]
```diff
@@ -373,26 +397,29 @@ func TestBlockVerify_PostForkBlock_PChainHeightChecks(t *testing.T) {
 	valState.GetCurrentHeightF = func(context.Context) (uint64, error) {
 		return pChainHeight, nil
 	}
+	valState.GetMinimumHeightF = func(context.Context) (uint64, error) {
+		return pChainHeight / 50, nil
+	}
```
Comment (abi87) [inline vms/proposervm/post_fork_block_test.go:402]: this missing was the reason of the wrong error assertion below

File: vms/proposervm/batched_vm_test.go [hunk 6]
```diff
@@ -211,10 +211,9 @@ func TestGetAncestorsPostForkOnly(t *testing.T) {
 			IDV:     ids.GenerateTestID(),
 			StatusV: choices.Processing,
 		},
-		BytesV:     []byte{1},
-		ParentV:    coreGenBlk.ID(),
-		HeightV:    coreGenBlk.Height() + 1,
-		TimestampV: coreGenBlk.Timestamp(),
```
Comment (abi87) [inline vms/proposervm/batched_vm_test.go:217]: these are inner blocks of would be post fork blocks. Their timestamp is irrelevant for proposerVM. Dropping it

File: vms/proposervm/vm_test.go [hunk 7]
```diff
@@ -210,6 +210,8 @@ func initTestProposerVM(
 	require.NoError(proVM.SetState(context.Background(), snow.NormalOp))
 	require.NoError(proVM.SetPreference(context.Background(), coreGenBlk.IDV))
 
+	proVM.Set(coreGenBlk.Timestamp())
```
Comment (abi87) [inline vms/proposervm/vm_test.go:213]: just set the proposerVM clock time as we exit

File: vms/proposervm/pre_fork_block_test.go [hunk 8]
```diff
@@ -268,15 +267,14 @@ func TestBlockVerify_PreFork_ParentChecks(t *testing.T) {
...
-		case bytes.Equal(b, prntCoreBlk.Bytes()):
-			return prntCoreBlk, nil
+		case bytes.Equal(b, parentCoreBlk.Bytes()):
+			return parentCoreBlk, nil
 		default:
 			return nil, database.ErrNotFound
 		}
 	}
 
-	proVM.Set(proVM.Time().Add(proposer.MaxBuildDelay))
```
Comment (abi87) [inline vms/proposervm/pre_fork_block_test.go:278]: These are pre fork blocks, setting proposerVM clock is pointless. Dropped

File: vms/proposervm/batched_vm_test.go (hunk 6 above)
Comment (patrick-ogrady) [reply vms/proposervm/batched_vm_test.go:217]: Thanks for taking this...caused a lot of confusion during the `MaxVerify/BuildDelay` change.

**Review by joshua-kim — APPROVED** [review]

**Pull request merged**